crates/runner/src/vm/instructions/call.rs

```diff
@@ -0,0 +1,287 @@
+//! CALL and RET instructions for the Cairo M VM.
+//!
+//! CALL instructions handle functions calls, creating new frames.
+//! There are relative and absolute function calls.
+//!
+//! The first argument, `off0` is the offset between the current frame pointer and the next frame pointer minus 2.
+//! In other words, `next_fp = fp + off0 + 2`.
+//! The second argument, `off1` is the destination offset.
+//!
+//! The function argumments are assumed to be stored
+//! in memory at [fp + off0 - N, ..., fp + off0 - 1], where N is the number of arguments.
+//!
+//! The function call is performed by:
+//! - Storing FP in memory at fp + off0.
+//! - Storing the return address in memory at fp + off0 + 1.
+//! - Updating FP for the new frame: fp + off0 + 2.
+//! - Updating PC to the function address based on off1.
+//!
+//! RET instructions returns control to the caller:
+//! - Restore FP from memory, stored at fp - 2.
+//! - Update PC to the return address, stored at fp - 1.
```

Comment:
Where are stored the return values?
When there is a ret, how one would access those?
Last time this was discussed, I thought it would be at the beginning/before the frame:
return values, then old fp then return pc